packages/quickjs/quickjs.0.1.0/opam

```diff
@@ -0,0 +1,44 @@
+opam-version: "2.0"
+synopsis:
+  "Bindings for QuickJS (a Javascript Engine to be embedded https://bellard.org/quickjs)"
+maintainer: ["David Sancho <dsnxmoreno@gmail.com>"]
+authors: ["David Sancho <dsnxmoreno@gmail.com>"]
+license: "MIT"
+homepage: "https://github.com/ml-in-barcelona/quickjs.ml"
+bug-reports: "https://github.com/ml-in-barcelona/quickjs.ml/issues"
+depends: [
+  "dune" {>= "3.8"}
+  "ocaml" {>= "5.0.0"}
+  "reason" {>= "3.10.0"}
+  "integers"
+  "ctypes"
+  "alcotest" {with-test}
+  "fmt" {with-test}
+  "odoc" {with-doc}
+  "ocaml-lsp-server" {with-test}
+  "ocamlformat" {= "0.26.1" & with-test}
+]
+build: [
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+dev-repo: "git+https://github.com/ml-in-barcelona/quickjs.ml.git"
```

Comment:
```suggestion
conflicts: [ "ocaml-option-bytecode-only" ]
dev-repo: "git+https://github.com/ml-in-barcelona/quickjs.ml.git"
```
to solve
```

#=== ERROR while compiling quickjs.0.1.0 ======================================#
# context              2.2.0~beta2~dev | linux/x86_32 | ocaml-base-compiler.5.1.1 | pinned(https://github.com/ml-in-barcelona/quickjs.ml/releases/download/0.1.0/quickjs-0.1.0.tbz)
# path                 ~/.opam/5.1/.opam-switch/build/quickjs.0.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p quickjs -j 39 @install
# exit-code            1
# env-file             ~/.opam/log/quickjs-7-ba2cab.env
# output-file          ~/.opam/log/quickjs-7-ba2cab.out
### output ###
# File "_build/default/libregexp__function_gen__Function_description__Functions", line 1, characters 0-0:
# Error: File unavailable: /home/opam/.opam/5.1/lib/ocaml/str/str.cmxa
# File "_build/default/libregexp__type_gen", line 1, characters 0-0:
# Error: No rule found for .bindings.objs/native/bindings.cmx
# File "_build/default/libregexp__type_gen", line 1, characters 0-0:
# Error: No rule found for .bindings.objs/native/bindings.o
# File "_build/default/libregexp__type_gen", line 1, characters 0-0:
# Error: No rule found for
# .bindings.objs/native/bindings__Libregexp__type_gen.cmx
# File "_build/default/libregexp__type_gen", line 1, characters 0-0:
# Error: No rule found for
# .bindings.objs/native/bindings__Libregexp__type_gen.o
# File "_build/default/libregexp__type_gen", line 1, characters 0-0:
# Error: No rule found for .bindings.objs/native/bindings__Type_description.cmx
# File "_build/default/libregexp__type_gen", line 1, characters 0-0:
# Error: No rule found for .bindings.objs/native/bindings__Type_description.o
```

Reply:
I would like to know the why (or if there's anything I can do in my package instead of the conflicts field)

Thanks for fixing it!

Reply:
The issue is that you are trying to compile and link a native code executable, and this fail if there is no native code compiler. The conflicts basically prevents to compile there.

I don't know by heart how to fix it, but it is not uncommon when you have c bindings like in your case. You probably need to tweak the way the c program is compiled to make it work. I'd suggest to do it in a future release and in the meantine you can announce the project. We have a community category and an announcement target on discuss.ocaml.org for this purpose

Reply:
Sounds good, would a simply `(modes native)` on the dune file work?

Reply:
It would still fail if there is only the bytecode compiler